doc/source/design/design-timestep.rst

```diff
@@ -0,0 +1,202 @@
+.. include:: ../roles.incl
+
+.. |tijk| replace:: t\ :sub:`i` :sup:`j,k`
+
+.. toctree::
+
+.. _Adaptive Time-Step Design:
+
+
+*************************
+Adaptive Time-Step Design
```

Comment:
This isn't essential, but we may want to acknowledge that subcycling also exists (e.g. in Grackle and radiative transfer). Maybe just a single sentence noting that this is used as a technique on a method-by-method basis.